### ISSUE TITLE
How to handle circular references

### ISSUE THREAD

**Issue by mr-at0s** [mfidemraizer/trackerdog#43]
Some objects may have circular references. How should we handle this? Maybe, throwing an exception if someone is registering such an object? Or any idea on this?

**Comment by mfidemraizer** [comment]
Hey!

I'm not sure what could be the best solution here. Ignoring them, throwing that exception...

Anyway, is this throwing exceptions? Probably I'm mistaken, but TrackerDog independently handles each property change.

**Comment by mr-at0s** [comment]
Currently, it does not throw an exception. It just runs into that cycle and never returns (until program is out of memory).

Potential solution: The user could provide an error callback that is called when a circular references is detected. But in each case a continuous check for circular references could be time consuming, wouldn't it?

Maybe, it is just a bad design in client's code (my) to allow potential circular references, e.g., tracking changes in a doubly linked list.

**Comment by mfidemraizer** [comment]
@mr-at0s It shouldn't be a bad design per se, because usually when you implement an `1-n` association, you may want a reference of both side in both sides:

```c#
public class Foo
{
      public IList<Bar> Bars { get; set; }
}

public class Bar
{
      public Foo Foo { get; set; }
}
```

Maybe a possible solution could be ignoring them with some attribute:

```c#
public class Foo
{
      public IList<Bar> Bars { get; set; }
}

public class Bar
{
      [IgnoreDataMember]
      public Foo Foo { get; set; }
}
```

Actually

What do you think about this?

**Comment by mfidemraizer** [comment]
Also, maybe the whole TrackerDog's configuration API could provide something like *track this but* (some given properties).

**Comment by mr-at0s** [comment]
Yes, `Ignore` might be a solution for such scenarios.

**Comment by mfidemraizer** [comment]
Do you need this feature already?

**Comment by mr-at0s** [comment]
Yes, but the performance issue (see https://github.com/mfidemraizer/trackerdog/issues/45) has got higher priority for us.

**Comment by mfidemraizer** [comment]
I see. Are you woking in a .NET Core app?

**Comment by mr-at0s** [comment]
No, these are standard Windows .Net Framework projects.
